### ISSUE TITLE
Time difference between 1.0.2 and 1.0.3

### ISSUE THREAD

**Issue by nheinemans** [rancher/os#1963]
RancherOS Version: 1.0.3

Running rancheros on vmware. Building vmware templates using packer.

Since rancheros 1.0.3, the time on the host is 2 hours ahead.

I'm running 1 environment with 2 hosts:
host1 - 1.0.2:
```
[rancher@csc91arm ~]$ date
Tue Jul  4 05:23:41 UTC 2017
```
host2 - 1.0.3:
```
[rancher@csy0ivkz ~]$ date
Tue Jul  4 07:23:27 UTC 2017
```
I've attached the contents of /var/lib/rancher/conf/cloud-config.yml on both hosts: [rancheros issue.txt](https://github.com/rancher/os/files/1121067/rancheros.issue.txt)

Open-vm-tools logs are identical on both systems:
```
system-docker logs open-vm-tools
> time="2017-07-04T07:18:47Z" level=error msg="symlink /usr/bin/ros /usr/bin/cloud-init-execute: file exists" 
> time="2017-07-04T07:18:47Z" level=error msg="symlink /usr/bin/ros /usr/bin/cloud-init-save: file exists" 
> time="2017-07-04T07:18:48Z" level=error msg="symlink /usr/bin/ros /usr/bin/dockerlaunch: file exists" 
> time="2017-07-04T07:18:48Z" level=error msg="symlink /usr/bin/ros /usr/bin/respawn: file exists" 
> time="2017-07-04T07:18:48Z" level=error msg="symlink /usr/bin/ros /usr/bin/system-docker: file exists" 
> time="2017-07-04T07:18:48Z" level=error msg="symlink /usr/bin/ros /usr/sbin/netconf: file exists" 
> time="2017-07-04T07:18:48Z" level=error msg="symlink /usr/bin/ros /usr/sbin/wait-for-docker: file exists" 
> time="2017-07-04T07:20:20Z" level=error msg="symlink /usr/bin/ros /usr/bin/cloud-init-execute: file exists" 
> time="2017-07-04T07:20:20Z" level=error msg="symlink /usr/bin/ros /usr/bin/cloud-init-save: file exists" 
> time="2017-07-04T07:20:20Z" level=error msg="symlink /usr/bin/ros /usr/bin/dockerlaunch: file exists" 
> time="2017-07-04T07:20:20Z" level=error msg="symlink /usr/bin/ros /usr/bin/respawn: file exists" 
> time="2017-07-04T07:20:20Z" level=error msg="symlink /usr/bin/ros /usr/bin/system-docker: file exists" 
> time="2017-07-04T07:20:20Z" level=error msg="symlink /usr/bin/ros /usr/sbin/netconf: file exists" 
> time="2017-07-04T07:20:20Z" level=error msg="symlink /usr/bin/ros /usr/sbin/wait-for-docker: file exists" 
```

Finally the reason we noticed this issue, in Rancher we can't see any performance graphs for containers on the 1.0.3 host. This is causing these errors in the rancher-agent log:
```
time="2017-07-04T07:42:51Z" level=error msg="Couldn't find container for id." error="token is expired" id=f42854003d148f76cd21308528e0b3328f15ac717d6496059b66891dd56cdb60 
time="2017-07-04T07:42:52Z" level=error msg="Couldn't find container for id." error="token is expired" id=f42854003d148f76cd21308528e0b3328f15ac717d6496059b66891dd56cdb60 

```

**Comment by SvenDowideit** [comment]
On my ESXi system, without using packer, I've started 2 vm's, and they both have the same time :(

**Comment by nheinemans** [comment]
After some more poking around, I found the following:
- When the ntp daemon starts, it should correct any time difference immediately (ntpd is running with the -g option). In 1.0.2, this works as expected.
- In 1.0.3, ntp daemon doesn't correct the time difference at all. In my case, the hardware clock is off by 2 hours. Although ntp daemon is started the same as in 1.0.2, the time stays incorrect.
- When I modify a 1.0.3 host to use the 1.0.2 version of the ntp daemon (container), the problem is solved. So the problem is likely caused by any changes in the rancher/os-base:1.0.3 image.

Since I can reproduce this issue consistently, I'd be more than willing to test the next 1.0.4-RC version.

**Comment by SvenDowideit** [comment]
@nheinemans what does `sudo ros service logs ntp`  output on the 2 systems?

I get 

```
[docker@rancher ~]$ sudo ros service logs ntp
ntp_1 | 19 Jul 02:35:52 ntpd[1]: ntpd 4.2.8p8@1.3265-o Tue Nov 22 03:00:43 UTC 2016 (1): Starting
ntp_1 | 19 Jul 02:35:52 ntpd[1]: Command line: ntpd --nofork -g
ntp_1 | 19 Jul 02:35:52 ntpd[1]: proto: precision = 0.238 usec (-22)
ntp_1 | 19 Jul 02:35:52 ntpd[1]: Listen and drop on 0 v6wildcard [::]:123
ntp_1 | 19 Jul 02:35:52 ntpd[1]: Listen and drop on 1 v4wildcard 0.0.0.0:123
ntp_1 | 19 Jul 02:35:52 ntpd[1]: Listen normally on 2 lo 127.0.0.1:123
ntp_1 | 19 Jul 02:35:52 ntpd[1]: Listen normally on 3 eth0 10.10.10.166:123
ntp_1 | 19 Jul 02:35:52 ntpd[1]: Listen normally on 4 lo [::1]:123
ntp_1 | 19 Jul 02:35:52 ntpd[1]: Listen normally on 5 eth0 [fe80::20c:29ff:fef8:5612%2]:123
ntp_1 | 19 Jul 02:35:52 ntpd[1]: Listening on routing socket on fd #22 for interface updates
ntp_1 | 19 Jul 02:35:47 ntpd[1]: receive: Unexpected origin timestamp 0xdd19440a.87db2959 does not match aorg 0000000000.00000000 from server@129.250.35.250 xmt 0xdd194402.fd2d9446
[docker@rancher ~]$ ros -v                                                                                          
ros version v1.0.2
```

and 

```
[root@v103 docker]# sudo ros service logs ntp
ntp_1 | 19 Jul 02:21:22 ntpd[1]: ntpd 4.2.8p10@1.3728-o Fri Jun  2 07:53:20 UTC 2017 (1): Starting
ntp_1 | 19 Jul 02:21:22 ntpd[1]: Command line: ntpd --nofork -g
ntp_1 | 19 Jul 02:21:22 ntpd[1]: proto: precision = 0.096 usec (-23)
ntp_1 | 19 Jul 02:21:22 ntpd[1]: Listen and drop on 0 v6wildcard [::]:123
ntp_1 | 19 Jul 02:21:22 ntpd[1]: Listen and drop on 1 v4wildcard 0.0.0.0:123
ntp_1 | 19 Jul 02:21:22 ntpd[1]: Listen normally on 2 lo 127.0.0.1:123
ntp_1 | 19 Jul 02:21:22 ntpd[1]: Listen normally on 3 eth0 10.10.10.162:123
ntp_1 | 19 Jul 02:21:22 ntpd[1]: Listen normally on 4 lo [::1]:123
ntp_1 | 19 Jul 02:21:22 ntpd[1]: Listen normally on 5 eth0 [fe80::20c:29ff:fe68:787c%2]:123
ntp_1 | 19 Jul 02:21:22 ntpd[1]: Listening on routing socket on fd #22 for interface updates
ntp_1 | 19 Jul 02:32:02 ntpd[1]: kernel reports TIME_ERROR: 0x41: Clock Unsynchronized
ntp_1 | 19 Jul 02:31:55 ntpd[1]: frequency error -10081 PPM exceeds tolerance 500 PPM
```

**Comment by SvenDowideit** [comment]
mmm, so the 1.0.3 build already has https://patchwork.ozlabs.org/patch/742869/, so we're actually using the build as it should be - and after running for a while, 1.0.2 also gets the same 

```
ntp_1 | 19 Jul 02:47:08 ntpd[1]: kernel reports TIME_ERROR: 0x41: Clock Unsynchronized
ntp_1 | 19 Jul 02:47:02 ntpd[1]: frequency error -8960 PPM exceeds tolerance 500 PPM
```

**Comment by SvenDowideit** [comment]
ok, so I'm going to make a script for the ntp service that runs `ntpd -gq` first, and then runs `ntpd` and we'll see how that goes.

**Comment by nheinemans** [comment]
I don't see the "frequency error" messages on my systems.

**Comment by SvenDowideit** [comment]
One thing to try (and if its good, i'll add it to 1.1 and 1.0.4) is to use the following in a cloud-init:

```
#cloud-config
rancher:
  services:
    ntp:
      command: /bin/start_ntp.sh
write_files:
  - container: ntp
    path: /bin/start_ntp.sh
    permissions: "0755"
    owner: root
    content: |
      #!/bin/sh
      set -ex
      echo "starting in one shot mode to fix large time differences"
      ntpd -gx
      echo "starting long running nptd"
      exec ntpd --nofork -g
```

**Comment by nheinemans** [comment]
This works for me. However, I had to change ntpd -gx to ntpd -gq.

**Comment by SvenDowideit** [comment]
#1997 will be in v1.1.0-rc2 and I'll backport it when ready.

**Comment by SvenDowideit** [comment]
#1997 was backported to v1.0.4

**Comment by nheinemans** [comment]
I just checked and the issue is resolved in 1.0.4.
